src/logic/editor_game_base.h

```diff
@@ -117,10 +118,26 @@ class EditorGameBase {
 	virtual void postload();
 	void load_graphics();
 	virtual void cleanup_for_load();
-	void set_loader_ui(UI::ProgressWindow*);
-	UI::ProgressWindow* get_loader_ui() {
-		return loader_ui_;
+
+	// Create a new loader UI
+	UI::ProgressWindow& create_loader_ui(const std::vector<std::string>& tipstexts,
+	                                     const std::string& background = std::string());
+
+	// Change the background image for the loader UI and remove the game tips
+	void change_loader_ui_background(const std::string& background);
+
+	// Set step text for the current loader UI if it's not nullptr.
+	void step_loader_ui(const std::string& text) const;
+
+#ifndef NDEBUG
```

Comment:
You can remove the `#ifndef` because having this in release builds as well won't hurt even if its unused

Reply:
Linters like cppcheck or clang tidy will complain about it if I leave it in for the release builds.